docs/IIPAddressRange.rst

```diff
@@ -0,0 +1,87 @@
+.. _IIPAddressRange:
+
+IIPAddressRange
+===============
+
+``IIPAddressRange`` is an interface defining a contract for all of Arcus' implementations of a consecutive a range of ``IPAddress`` objects. It implements both ``IFormattable`` and ``IEnumerable<IPAddress>``
+
+.. caution:: ``IIPAddressRange`` implements ``IEnumerable<IPAddress>``, this means that you should pay particular attention when you may be iterating over large ranges. Such as the full set of IPv6 addresses. That'll take a while. A long while. It isn't recommended.
+
+.. hint:: When dealing with more than one ``IPAddress`` or multiple implementations of ``IIPAddressRange`` unless otherwise explicitly stated their ``AddressFamily`` , or equivalent properties, **must** match.
+
+.. hint:: ``AddressFamily`` unless otherwise explicitly stated are expected to be either ``InterNetwork`` or ``InterNetworkV6``.
+
+``IIPAddressRange`` is implemented by :ref:`AbstractIPAddressRange` , :ref:`IPAddressRange` , and :ref:`Subnet`
+
+Functionality Promises
+----------------------
+
+Properties
+^^^^^^^^^^
+
+``IIPAddressRange`` has a handful of hopefully useful properties for your use
+
+:``AddressFamily`` AddressFamily: The family of the Address Range. You'll most likely encounter ``InterNetwork`` or ``InterNetworkV6``
+:``IPAddress`` Head: The first ``IPAddress`` within the range
+:``bool`` IsIPv4: Returns ``true`` iff the range is IPv4
+:``bool`` IsIPv6: Returns ``true`` iff the range is IPv6
+:``bool`` IsSingleIP: Returns ``true`` iff the range is comprised of only a single ``IPAddress``
+:``BigInteger`` Length: The number of ``IPAddress`` within the range
+:``IPAddress`` Tail: The last ``IPAddress`` within the range
+
+
+Set Based Operations
+^^^^^^^^^^^^^^^^^^^^
+
+Inherently a ``IIPAddressRange`` is simply a range of consecutive ``IPAddress`` objects, as such there are some set based operations available.
+
+HeadOverlappedBy
+~~~~~~~~~~~~~~~~
+
+``HeadOverlappedBy`` will return ``true`` if the ``head`` of ``this IIPAddressRange`` is within the ``IIPAddressRange that``
+
+.. code-block:: c#
+
+   bool HeadOverlappedBy(IIPAddressRange that);
+
+TailOverlappedBy
+~~~~~~~~~~~~~~~~
+
+``TailOverlappedBy`` will return ``true`` if the ``tail`` of ``this IIPAddressRange`` is within the ``IIPAddressRange that``
+
+.. code-block:: c#
+
+   bool TailOverlappedBy(IIPAddressRange that);
+
+Overlaps
+~~~~~~~~
+
+``Overlaps`` will return ``true`` if the ``head`` or ``tail`` of ``IIPAddressRange that`` is within the ``this IIPAddressRange``
+
+.. code-block:: c#
+
+   bool Overlaps(IIPAddressRange that);
+
+Touches
+~~~~~~~
+
+``Touches`` will return ``true`` if the ``tail`` of ``this IIPAddressRange`` is followed consecutively by the ``head`` of ``IIPAddressRange that`` , or if the ``tail`` of ``IIPAddressRange that`` is followed consecutively by the ``head`` of ``this IIPAddressRange`` without any additional ``IPAddress`` objects in between.
```

Comment:
Found 10 instances of ` `` , ` and removed white space

```
D:\dev\arcus\docs\IIPAddressRange.rst
  10,164:  otherwise explicitly stated their ``AddressFamily`` , or equivalent properties, **must** match.


D:\dev\arcus\docs\IPAddressRange.rst
  8,136: The ``IPAddressRange`` class extends :ref:`AbstractIPAddressRange` and implements :ref:`IIPAddressRange` , ``IEquatable<IPAddressRange>`` , ``IComparable<IPAddressRange>`` , ``IFormattable`` , and ``IEnumerable<IPAddress>``.

  8,170: The ``IPAddressRange`` class extends :ref:`AbstractIPAddressRange` and implements :ref:`IIPAddressRange` , ``IEquatable<IPAddressRange>`` , ``IComparable<IPAddressRange>`` , ``IFormattable`` , and ``IEnumerable<IPAddress>``.

  8,189: The ``IPAddressRange`` class extends :ref:`AbstractIPAddressRange` and implements :ref:`IIPAddressRange` , ``IEquatable<IPAddressRange>`` , ``IComparable<IPAddressRange>`` , ``IFormattable`` , and ``IEnumerable<IPAddress>``.


D:\dev\arcus\docs\Subnet.rst
  17,121: ly candidates are direct construction with a ``new`` , the use of a static factory method on the ``Subnet`` class, or the use of sub-set of static factory methods that handle parsing of strings. Most of the factory methods have a "try" style safe alt
  124,72: The ``Subnet`` implements :ref:`IIPAddressRange` , ``IEquatable<Subnet>`` , ``IComparable<Subnet>`` , ``IFormattable`` , and ``IEnumerable<IPAddress>`` , and there by contains all the expected functionality its inheritance.

  124,98: The ``Subnet`` implements :ref:`IIPAddressRange` , ``IEquatable<Subnet>`` , ``IComparable<Subnet>`` , ``IFormattable`` , and ``IEnumerable<IPAddress>`` , and there by contains all the expected functionality its inheritance.

  124,117: The ``Subnet`` implements :ref:`IIPAddressRange` , ``IEquatable<Subnet>`` , ``IComparable<Subnet>`` , ``IFormattable`` , and ``IEnumerable<IPAddress>`` , and there by contains all the expected functionality its inheritance.

  124,150: The ``Subnet`` implements :ref:`IIPAddressRange` , ``IEquatable<Subnet>`` , ``IComparable<Subnet>`` , ``IFormattable`` , and ``IEnumerable<IPAddress>`` , and there by contains all the expected functionality its inheritance.

  153,83: hown that ``192.168.1.0/8`` contains ``192.168.0.0`` , but as expected ``192.168.1.0/8`` does not contain ``255.0.0.0/8``

```